examples/cryptocurrency-advanced/backend/src/transactions.rs

```diff
@@ -65,44 +66,62 @@ impl From<Error> for ExecutionError {
     }
 }
 
-transactions! {
-    /// Transaction group.
-    pub WalletTransactions {
-
-        /// Transfer `amount` of the currency from one wallet to another.
-        struct Transfer {
-            /// `PublicKey` of receiver's wallet.
-            to:      &PublicKey,
-            /// Amount of currency to transfer.
-            amount:  u64,
-            /// Auxiliary number to guarantee [non-idempotence][idempotence] of transactions.
-            ///
-            /// [idempotence]: https://en.wikipedia.org/wiki/Idempotence
-            seed:    u64,
-        }
+/// Transfer `amount` of the currency from one wallet to another.
+#[derive(Serialize, Deserialize, Clone, Debug, ProtobufConvert)]
+#[protobuf_convert("proto::Transfer")]
+pub struct Transfer {
+    /// `PublicKey` of receiver's wallet.
+    pub to: PublicKey,
+    /// Amount of currency to transfer.
+    pub amount: u64,
+    /// Auxiliary number to guarantee [non-idempotence][idempotence] of transactions.
+    ///
+    /// [idempotence]: https://en.wikipedia.org/wiki/Idempotence
+    pub seed: u64,
+}
 
-        /// Issue `amount` of the currency to the `wallet`.
-        struct Issue {
-            /// Issued amount of currency.
-            amount:  u64,
-            /// Auxiliary number to guarantee [non-idempotence][idempotence] of transactions.
-            ///
-            /// [idempotence]: https://en.wikipedia.org/wiki/Idempotence
-            seed:    u64,
-        }
+/// Issue `amount` of the currency to the `wallet`.
+#[derive(Serialize, Deserialize, Clone, Debug, ProtobufConvert)]
+#[protobuf_convert("proto::Issue")]
+pub struct Issue {
+    /// Issued amount of currency.
+    pub amount: u64,
+    /// Auxiliary number to guarantee [non-idempotence][idempotence] of transactions.
+    ///
+    /// [idempotence]: https://en.wikipedia.org/wiki/Idempotence
+    pub seed: u64,
+}
 
-        /// Create wallet with the given `name`.
-        struct CreateWallet {
-            /// Name of the new wallet.
-            name:    &str,
-        }
-    }
+/// Create wallet with the given `name`.
+#[derive(Serialize, Deserialize, Clone, Debug, ProtobufConvert)]
+#[protobuf_convert("proto::CreateWallet")]
+pub struct CreateWallet {
+    /// Name of the new wallet.
+    pub name: String,
+}
+
+/// Transaction group.
+#[derive(Serialize, Deserialize, Clone, Debug, TransactionSet)]
+pub enum WalletTransactions {
```

Comment:
I believe later we could hide all this complexity in Protobuf codegen so that it generates correct Rust structures and enumerations from a *.proto file describing the service.